src/EmailReporter.ts

```diff
@@ -0,0 +1,66 @@
+import { readFileSync } from 'fs'
+
+import * as nodemailer from "nodemailer";
+import * as openpgp from 'openpgp';
+
+export interface EmailConfig {
+	from: string,
+	to: string[],
+	gpgpubkey?: string,
+	transporter: any
+}
+
+export class EmailReporter {
+	maybePubkey: openpgp.Key | undefined;
+	transporter: nodemailer.Transporter;
+	from: string;
+	to: string[];
+
+	constructor(config: EmailConfig) {
+		if (config.transporter["dkim"]) {
+			config.transporter["dkim"]["privateKey"] = readFileSync(config.transporter["dkim"]["privateKey"]).toString()
+		}
+		const transporter = nodemailer.createTransport(config.transporter);
+
+		if (config.gpgpubkey) {
+			openpgp.readKey({ armoredKey: readFileSync(config.gpgpubkey).toString() })
+				.then((p) => this.maybePubkey = p);
+		}
+
+		this.transporter = transporter;
+		this.from = config.from;
+		this.to = config.to;
+		//logger.info(`✅ registering email reporter from ${this.from} to ${this.to}.`)
+	}
+
+	async maybeEncrypt(message: string): Promise<string> {
+		if (this.maybePubkey) {
+			const enc = await openpgp.encrypt({
+				message: await openpgp.createMessage({ text: message }),
+				encryptionKeys: this.maybePubkey,
+			})
+			// Todo: fix?
```

Comment:
do you know what this is about?

Reply:
It used to complain about line 43, where enc was returned (by default) as a Webstream<String>. 
This silenced the complaints, but I had not tested its functionality yet.